controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -632,11 +632,30 @@ public void initializeStreamWriters(final EventStreamClientFactory clientFactory
     @VisibleForTesting
     <T> CompletableFuture<T> addIndexAndSubmitTask(ControllerEvent event, Supplier<CompletableFuture<T>> futureSupplier) {
         String id = UUID.randomUUID().toString();
+        // We first add index and then call the metadata update.
+        // In case, while trying to perform a metadata update, if we get connection exception or write conflict exception 
```

Comment:
Here is slight rephrase of this comment:

```
//  While trying to perform a metadata update, upon getting a connection exception or a write conflict exception 
// (which can also occur if we had retried on a store exception), we will still post the event because we
//  don't know whether our update succeeded. Posting the event is harmless, though. If the update
// has succeeded, then the event will be used for processing. If the update had failed, then the event
// will be discarded. We will throw the exception that we thrown from running futureSupplier.

```